app/services/order_service.py

```diff
@@ -0,0 +1,182 @@
+from uuid import UUID
+from datetime import datetime, timezone
+from decimal import Decimal
+from fastapi import HTTPException, status
+from sqlalchemy.orm import Session,joinedload
+from app.models import Order,Product,Status,OrderProduct
+from app.schemas import CreateOrderRequestModel, CreateOrderResponseModel, GetOrderResponseModel, OrderProductBaseModel, UpdateOrderStatusResponseModel
+
+
+class OrderService:
+    def __init__(self, db: Session):
+        self.db = db
+
+    def create_order(self, user_id: UUID, order_request: CreateOrderRequestModel) -> CreateOrderResponseModel:
```

Comment:
No need for all this complexity and validation, that is the reason why we use ORM, instead what we should do is only pass the order DB model with all its order product, and in a single line, you can create the order, please read this [example](https://docs.sqlalchemy.org/en/20/orm/extensions/orderinglist.html).
long story short you should update your order DB model to use a relationship something like
```
class Order(Base):
    __tablename__ = 'orders'
    
    # other fileds
    order_products = relationship("OrderProduct", back_populates="order")
```

and in order product model something like 
```
    order = relationship("Order", back_populates="order_products")
```

please update all other DB functions to use this approach

Reply:
I see what you did, usually all these validations we added it to a trigger function in DB, and we only insert the order itself, anyway Don't worry if you don't fully understand this or know how to implement it right now.

Reply:
You can revert the previous changes, but you should also update the product stock after creating the order.